updater/lib/dependabot/updater/security_update_helpers.rb

```diff
@@ -181,5 +181,52 @@ def security_update_not_possible_message(checker, latest_allowed_version, confli
         end
       end
     end
+
+    module PullRequestHelpers
+      extend T::Sig
+      extend T::Helpers
+
+      sig { returns(Dependabot::Service) }
+      attr_reader :service
+
+      abstract!
+
+      sig { params(notices: T.nilable(T::Array[Dependabot::Notice])).void }
+      def record_warning_notices(notices)
+        return if !notices || notices.empty?
+
+        # Find unique warning notices which are going to be shown on insight page.
+        warn_notices = unique_warn_notices(notices)
+
+        warn_notices.each do |notice|
+          # If alert is enabled, sending the deprecation notice to the service for showing on the UI insight page
+          send_alert_notice(notice) if notice.show_alert
+        end
+        rescue StandardError => e
+          Dependabot.logger.error(
+            "Failed to send notice warning: #{e.message}"
+          )
+      end
+
+      private
+
+      # Resurns unique warning notices which are going to be shown on insight page.
+      sig { params(notices: T::Array[Dependabot::Notice]).returns(T::Array[Dependabot::Notice]) }
+      def unique_warn_notices(notices)
+        notices
+          .select { |notice| notice.mode == Dependabot::Notice::NoticeMode::WARN }
+          .uniq { |notice| [notice.type, notice.package_manager_name] }
```

Comment:
is it possible to have multiple notices of the same type and from the same package manager, but with different titles or descriptions?  for example if/when bundler v2 is deprecated that could conceivably be the same notice type but with a different message?

Reply:
Actually for the deprecation it is not possible for the same job because we are getting the current version and checking the deprecation. So they can only have one deprecation notice for package manager. As for in general there can be notices that has same type, for same package manager with different title and description. I think that is possible but it may be better to go from narrow filtering then depends to the warning to expand it. What do you think?

Reply:
👍 sounds good